substrate-processor/src/chain.ts

```diff
@@ -34,7 +40,10 @@ interface BlockInfo {
     runtimeVersion: SpecVersion | SubstrateRuntimeVersion
 }
 
-
+/**
+ * Keeps a mapping of {@link Chain} linked to the blockchain's height, to encapsulate metadata in relation to the 
```

Comment:
Sounds confusing to me. The Manager keeps track of spec versions and maps them to the segments of blocks for which the given spec version is constant. Such block segments (called Chains by Eldar here -- not really a good name imo) have well-defined metadata for events, extrinsics and storage items, so it is possible to decode them.

Reply:
Definitely better wording.